Description: Use seek whenever possible to skip parts of a file.
Author: Michael Meskes <meskes@debian.org>

--- bsdmainutils/usr.bin/hexdump/display.c	2010-01-21 13:12:52.000000000 +0100
+++ bsdmainutils/usr.bin/hexdump/display.c	2010-01-21 13:13:20.000000000 +0100
@@ -383,11 +382,9 @@
 			return;
 		}
 	}
+	if (fseeko(stdin, skip, SEEK_SET)) {
 	if (S_ISREG(sb.st_mode)) {
-		if (fseeko(stdin, skip, SEEK_SET))
 			err(1, "%s", fname);
-		address += skip;
-		skip = 0;
 	} else {
 		for (cnt = 0; cnt < skip; ++cnt)
 			if (getchar() == EOF)
@@ -395,4 +392,9 @@
 		address += cnt;
 		skip -= cnt;
 	}
+	}
+	else {
+		address += skip;
+		skip = 0;
+	}
 }
